### ISSUE TITLE
freezing video and skipping

### ISSUE THREAD

**Issue by omeringen** [Zren/mpvz#4]
Hi,
I am browsing files(watching videos) on my android phone via KDE connect.
mpv works fine but mpvz is freezing after a while(just like there is no cache) and skipping to the next video.

How can i diagnose the difference between mpv and mpvz to help ?

**Comment by Zren** [comment]
Huh, didn't realize KDE Connect would stream the file without fully copying over first.

I've implemented the "Total Cache" stat in the Playback Info (`Tab`). I've also implemented cache range indicators in the seekbar (Issue #5).

![2021-07-25___17-51-13](https://user-images.githubusercontent.com/416367/126914557-ac56e4cc-ae54-47f0-86f5-51f38257a5f9.png)

They might not be perfect at diagnosing the issue though since it sounds like the connection disconnects or something.

**Comment by omeringen** [comment]
> They might not be perfect at diagnosing the issue though since it sounds like the connection disconnects or something.

I dont think that it's a connection issue because mpv works fine everytime i try. This happens especially on long videos. Thanks for implementations. :+1:  I am going to try, might take long time to find a way to explain. :smile:

**Comment by omeringen** [comment]
![Screenshot_20210727_110614](https://user-images.githubusercontent.com/22679786/127120149-f01584fe-729e-4e2c-adff-a7154ea454e3.png)

This happens most of the time (couldn't remember any successfully watched big file). I directly open file with mpv afterwards to finish the file from start to end, no issues. 

I am not a programmer or expert but mpvz is not using any modified libmpv, so this leads me to suspect something about "mpvz only" features. Also cache info at seekbar is not accurate.

bTw thanks for developing. We needed something to replace Bomi.

**Comment by Zren** [comment]
Hmm, the thumbnail generator does open up a 2nd libmpv filestream, so it makes sense. KDE Connect might only handle 1 file at a time? I'll try disabling it for "remote" streams somehow.

**Comment by Zren** [comment]
Taking a look at bomi's code, it seems it also created a mpv instance for the thumbnail.

* https://github.com/ashinan/bomi/blob/master/src/bomi/video/videopreview.cpp#L32
* https://github.com/ashinan/bomi/blob/master/src/bomi/video/videopreview.hpp

**Comment by omeringen** [comment]
> Hmm, the thumbnail generator does open up a 2nd libmpv filestream, so it makes sense. KDE Connect might only handle 1 file at a time? I'll try disabling it for "remote" streams somehow.

Thanks.

> Taking a look at bomi's code, it seems it also created a mpv instance for the thumbnail.
> 
>     * https://github.com/ashinan/bomi/blob/master/src/bomi/video/videopreview.cpp#L32
> 
>     * https://github.com/ashinan/bomi/blob/master/src/bomi/video/videopreview.hpp

I haven't tested it with Bomi. Not maintained anymore and AUR package wasn't working last time i checked.
